src/dos/dos.cpp

```diff
@@ -43,120 +45,152 @@ uint16_t countryNo = 0;
-        COUNTRY::Brazil,        COUNTRY::Turkey, COUNTRY::Arabic,
-        COUNTRY::Taiwan,        COUNTRY::Israel,
-};
-
-std::array<uint16_t, 8> countries_date_separator_period = {
-        COUNTRY::Russia,  COUNTRY::France,  COUNTRY::Switzerland,
-        COUNTRY::Austria, COUNTRY::Norway,  COUNTRY::Germany,
-        COUNTRY::China,   COUNTRY::Finland,
-};
+static constexpr uint8_t DATE_MDY   = 0;
+static constexpr uint8_t DATE_DMY   = 1;
+static constexpr uint8_t DATE_YMD   = 2;
 
-std::array<uint16_t, 3> countries_time_12hour_clock = {
-        COUNTRY::United_States,
-        COUNTRY::Latin_America,
-        COUNTRY::Australia,
-};
+static constexpr uint8_t TIME_12H   = 0;
+static constexpr uint8_t TIME_24H   = 1;
 
-std::array<uint16_t, 1> countries_time_separator_comma = {
-        COUNTRY::Switzerland,
-};
+static constexpr uint8_t SEP_SPACE  = 0x20; // ( )
+static constexpr uint8_t SEP_APOST  = 0x27; // (')
+static constexpr uint8_t SEP_COMMA  = 0x2c; // (,)
+static constexpr uint8_t SEP_DASH   = 0x2d; // (-)
+static constexpr uint8_t SEP_PERIOD = 0x2e; // (.)
+static constexpr uint8_t SEP_SLASH  = 0x2f; // (/)
+static constexpr uint8_t SEP_COLON  = 0x3a; // (:)
 
-std::array<uint16_t, 4> countries_time_separator_period = {
-        COUNTRY::Italy,
-        COUNTRY::Denmark,
-        COUNTRY::Sweden,
-        COUNTRY::Finland,
-};
+typedef struct CountryInfo {
+	COUNTRY country_number;
+	uint8_t date_format;
+	uint8_t date_separator;
+	uint8_t time_format;
+	uint8_t time_separator;
+	uint8_t thousands_separator;
+	uint8_t decimal_separator;
+} CountryInfo;
 
-std::array<uint16_t, 7> countries_decimal_separator_comma = {
-        COUNTRY::United_States, COUNTRY::United_Kingdom, COUNTRY::Australia,
-        COUNTRY::Japan,         COUNTRY::South_Korea,    COUNTRY::China,
-        COUNTRY::Taiwan,
-};
-
-std::array<uint16_t, 14> countries_decimal_separator_period = {
-        COUNTRY::Latin_America, COUNTRY::Netherlands, COUNTRY::Spain,
-        COUNTRY::Yugoslavia,    COUNTRY::Italy,       COUNTRY::Romania,
-        COUNTRY::Denmark,       COUNTRY::Sweden,      COUNTRY::Germany,
-        COUNTRY::Brazil,        COUNTRY::Portugal,    COUNTRY::Iceland,
-        COUNTRY::Arabic,        COUNTRY::Israel,
+// Values here reflect the current KDE/Linux system settings - they will probably not produce 100% same
+// result as old MS-DOS systems, but should at least provide reasonably consistent user experience with
+// certain host operating systems.
+static const std::vector<CountryInfo> COUNTRY_INFO = {
+	//                        | Date fmt | Date separ | Time fmt | Time separ | 1000 separ | Dec separ  |
+//	{ COUNTRY::None           , DATE_DMY , SEP_PERIOD , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_PERIOD }, // C
+	{ COUNTRY::United_States  , DATE_MDY , SEP_SLASH  , TIME_12H , SEP_COLON  , SEP_COMMA  , SEP_PERIOD }, // en_US
+	{ COUNTRY::Candian_French , DATE_YMD , SEP_DASH   , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_COMMA  }, // fr_CA
```

Comment:
We can hide the storage format inside a lookup function, so users simply get back a `CountryInfo` instead of having to crawl the layout.  The other benefit is that we can return a const-reference (instead of copying the data fields).

Regarding the vector, its content is populated at runtime (which is good for when its content might change at runtime or we're loading stuff dynamically); in this case I think this content is locked in stone, without any need for dynamic adjustment at runtime - so ideally we could make it all constexpr (either a standard C array or a std::array). 

Here's a go at that:

``` c++
static const CountryInfo& LookupCountryInfo(const uint16_t country_number) {

	// Values here reflect the current KDE/Linux system settings - they will probably not produce 100% same
	// result as old MS-DOS systems, but should at least provide reasonably consistent user experience with
	// certain host operating systems.

	static constexpr CountryInfo COUNTRY_INFO[] = {
		//                        | Date fmt | Date separ | Time fmt | Time separ | 1000 separ | Dec separ  |
		//{ COUNTRY::None         , DATE_DMY , SEP_PERIOD , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_PERIOD }, // C
		{ COUNTRY::United_States  , DATE_MDY , SEP_SLASH  , TIME_12H , SEP_COLON  , SEP_COMMA  , SEP_PERIOD }, // en_US
		{ COUNTRY::Candian_French , DATE_YMD , SEP_DASH   , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_COMMA  }, // fr_CA
		 ...
		{ COUNTRY::Uzbekistan     , DATE_DMY , SEP_SLASH  , TIME_24H , SEP_COLON  , SEP_SPACE  , SEP_COMMA  }, // uz_UZ
		//                        | Date fmt | Date separ | Time fmt | Time separ | 1000 separ | Dec separ  |
	};

	for (const auto& country : COUNTRY_INFO)
	{
		if (country.country_number == country_number)
			return country;
	}
	return COUNTRY_INFO[0];
}
```

Then in `DOS_SetCountry`:

``` c++
void DOS_SetCountry(const uint16_t country_number)
{
	if (dos.tables.country == nullptr)
		return;

	const auto country_info = LookupCountryInfo(country_number);
	// or
	// const CountryInfo & country_info = LookupCountryInfo(country_number);

	...
}
```